knees4bees/rancid-tomatillos
# Feature/api calls

## Is this a fix, feature, or refactor?
Feature

## What is the change?
* Adds linting and resolves several linting warnings/errors.
* Adds error handling for network requests.
* Creates and styles new `Error` component.

## What does it fix?
n/a

## Where should the reviewer start?
Look at `App.js` and `utilities.js` and note the new flow (`utilities` now just holds the fetch calls themselves; two new state properties are in `App.js`; there is a new `renderError` method that calls a new `Error` component). There is definitely room for refactoring here -- which is something we'll want to do on the PR we tag the instructors on (see the project spec). 

## How should this be tested?
* Linting: run `npm run lint`. If the linter causes the app to crash (this may happen when you run `npm start`), you may need to navigate to `node_modules` > `react-scripts` > `config` > `webpack.config.js`, look for the `ESLintPlugin` section near line 750, and set `failOnError: false`. (There is probably a better way to do this??)
* Error handling: first confirm that the happy path is still happy. Then uncomment line 57 in `App.js` and change the status code in the parentheses to whatever you'd like. An appropriate error message should appear on the screen.


## Relevant ScreenShots
